src/main/java/cam72cam/immersiverailroading/render/item/TrackExchangerModel.java

```diff
@@ -0,0 +1,188 @@
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+		if (world == null) world = Minecraft.getMinecraft().world;
+		
+		this.renderer = RENDERER;
+		String track = ItemTrackExchanger.get(stack);
+		this.info = INFO.withTrack(track);
+		this.info2 = null;
+		if (Minecraft.getMinecraft().objectMouseOver != null && Minecraft.getMinecraft().objectMouseOver.typeOfHit == RayTraceResult.Type.BLOCK) {
+			TileRailBase railSlave = TileRailBase.get(world, Minecraft.getMinecraft().objectMouseOver.getBlockPos());
+			if (railSlave != null) {
+				TileRail rail = railSlave.getParentTile();
+				if (rail != null) {
+					this.info2 = INFO.withTrack(rail.info.settings.track);
+				}
+			}
+		}
+	}
+
+	@Override
+	public List<BakedQuad> getQuads(IBlockState state, EnumFacing side, long rand) {
+		if (info == null) {
+			return new ArrayList<BakedQuad>();
+		}
+	
+		if (this.renderer != null) {
+			GLBoolTracker tex = new GLBoolTracker(GL11.GL_TEXTURE_2D, this.renderer.hasTexture());
+			GLBoolTracker cull = new GLBoolTracker(GL11.GL_CULL_FACE, false);
+
+			GL11.glPushMatrix();
+			this.renderer.bindTexture();
+			this.renderer.draw();
+			this.renderer.restoreTexture();
+			
+			GL11.glScaled(0.01, 0.01, 0.01);
+			GL11.glRotated(90, 1, 0, 0);
+			
+			GL11.glTranslated(-15.15, 0.75, -8.75);
+			RailBaseRender.draw(this.info);
+			RailBuilderRender.renderRailBuilder(this.info);
+			
+			if (this.info2 != null) {
+				GL11.glTranslated(-22.05, 0, 0);
+				RailBaseRender.draw(this.info2);
+				RailBuilderRender.renderRailBuilder(this.info2);
+			}
+			
+			GL11.glPopMatrix();
+			
+			tex.restore();
+			cull.restore();
+			
+			this.renderer = null;
+		}
+		
+		return new ArrayList<BakedQuad>();
+	}
+
+	@Override
+	public boolean isAmbientOcclusion() {
+		return true;
+	}
+
+	@Override
+	public boolean isGui3d() {
+		return true;
+	}
+
+	@Override
+	public boolean isBuiltInRenderer() {
+		return false;
+	}
+
+	@Override
+	public TextureAtlasSprite getParticleTexture() {
+		return null;
+	}
+	
+	public class ItemOverrideListHack extends ItemOverrideList {
+		public ItemOverrideListHack() {
+			super(new ArrayList<ItemOverride>());
+		}
+
+		@Override
+		public IBakedModel handleItemState(IBakedModel originalModel, ItemStack stack, @Nullable World world, @Nullable EntityLivingBase entity) {
+			return new TrackExchangerModel(stack, world);
+		}
+	}
+
+	@Override
+	public ItemOverrideList getOverrides() {
+		return new ItemOverrideListHack();
+	}
+
+	@Override
+	public Pair<? extends IBakedModel, Matrix4f> handlePerspective(TransformType cameraTransformType) {
+		Pair<? extends IBakedModel, Matrix4f> defaultVal = ForgeHooksClient.handlePerspective(this, cameraTransformType);
+		switch (cameraTransformType) {
+		case THIRD_PERSON_LEFT_HAND:
+		case THIRD_PERSON_RIGHT_HAND:
+			return Pair.of(defaultVal.getLeft(), new Matrix4().scale(1.5, 1.5, 1.5).translate(0.5, 0.5, 0.5).toMatrix4f());
```

Comment:
Right/Left should not use the same settings, also what happens when the user clicks with the track in the off hand?